### PR TITLE
Optimized get and extra docs

### PR DESCRIPTION
**Pull request by cormullion**
includes code from https://github.com/JuliaGraphics/ColorSchemes.jl/issues/91

extra section in docs for https://github.com/JuliaGraphics/ColorSchemes.jl/issues/90

### CODE REVIEW

**Comment by cormullion** [comment]
I don't know how to preview documentation in a PR ... :)

Anyway, I added a section which looks like this:

Screenshot 2022-04-22 at 11 00 27

which contains some comments about Lightness and perceptual uniformity.

**Comment by cormullion** [comment]
Documentation preview at 

https://juliagraphics.github.io/ColorSchemes.jl/previews/PR92/goodpractice/

File: data/colorbrewerschemes.jl [hunk 1]
```diff
@@ -2485,3 +2485,345 @@ loadcolorscheme(:PiYG_11, [
...
+    RGB(0.8392156862745098, 0.3764705882352941, 0.30196078431372547,)
+    RGB(0.9568627450980393, 0.6470588235294118, 0.5098039215686274,)
+    RGB(0.9921568627450981, 0.8588235294117647, 0.7803921568627451,)
+    RGB(0.9686274509803922, 0.9686274509803922, 0.9686274509803922,)
+    RGB(0.8196078431372549, 0.8980392156862745, 0.9411764705882353,)
+    RGB(0.5725490196078431, 0.7725490196078432, 0.8705882352941177,)
+    RGB(0.2627450980392157, 0.5764705882352941, 0.7647058823529411,)
+    RGB(0.12941176470588237, 0.4, 0.6745098039215687,)
+    RGB(0.0196078431372549, 0.18823529411764706, 0.3803921568627451,)
+      ], "colorbrewer", "perceptually uniform sequential")
```
Comment (stevengj) [inline data/colorbrewerschemes.jl:2701]: sequential or diverging?

File: src/ColorSchemes.jl [hunk 2]
```diff
@@ -281,20 +282,28 @@ function get(cscheme::ColorScheme, x::AllowedInput, rangemode::Symbol)
...
+        rangescale : (zero(rangescale[1]), oneunit(rangescale[2]))
+    scaleby = (length(cscheme) - 1) / (rangemax - rangemin)
+    return map(X) do x
+        xclamp = clamp(x, rangemin, rangemax)
+        before_fp = (xclamp - rangemin) * scaleby + 1
+        before = round(Int, before_fp, RoundDown)
+        after =  min(before + 1, length(cscheme))
+        cpt = before_fp - before
+        #  blend between the two colors adjacent to the point
+        @inbounds fast_weighted_color_mean(1 .- cpt, cscheme.colors[before], cscheme.colors[after])
```
Comment (stevengj) [inline src/ColorSchemes.jl:303]:
```suggestion
        @inbounds fast_weighted_color_mean(1 - cpt, cscheme.colors[before], cscheme.colors[after])
```
(`cpt` is a scalar)

File: data/colorbrewerschemes.jl (hunk 1 above)
Comment (cormullion) [reply data/colorbrewerschemes.jl]: Fixed.

File: src/ColorSchemes.jl (hunk 2 above)
Comment (cormullion) [reply src/ColorSchemes.jl]: Fixed.

**Pull request merged**